### ISSUE TITLE
train checkpoint it's very big 5G eveytime how can make smaler in train

### ISSUE THREAD

**Issue by lpscr** [SWivid/F5-TTS#208]
@SWivid i wonder if i can change this part checkpoint if curse something if i do ?

https://github.com/SWivid/F5-TTS/blob/99190a03cb473f1d139b7f282e681bbd60c7fd0e/model/trainer.py#L119-L135

When I save my model's checkpoint, it saves as 5GB, 

The issue is that if I have for example 10 checkpoints, I end up using 50GB ... i of disk space, and I often run out of space and if i make more train need more space . Do you have any ideas on how to make these checkpoints smaller? this safe ?
to keep only the **ema_model_state_dict** 1.3GB.

so only last model save 5G and check point save only **ema_model_state_dict** 1.3GB what you think ?

```
    def save_checkpoint(self, step, last=False):
        self.accelerator.wait_for_everyone()
        if self.is_main:
            if last:
               checkpoint = dict(
                model_state_dict=self.accelerator.unwrap_model(self.model).state_dict(),
                optimizer_state_dict=self.accelerator.unwrap_model(self.optimizer).state_dict(),
                ema_model_state_dict=self.ema_model.state_dict(),
                scheduler_state_dict=self.scheduler.state_dict(),
                step=step,
                )
            else:
               checkpoint = dict(ema_model_state_dict=self.ema_model.state_dict())

            if not os.path.exists(self.checkpoint_path):
                os.makedirs(self.checkpoint_path)
            if last:
                self.accelerator.save(checkpoint, f"{self.checkpoint_path}/model_last.pt")
                print(f"Saved last checkpoint at step {step}")
            else:
                self.accelerator.save(checkpoint, f"{self.checkpoint_path}/model_{step}.pt")
```

**Comment by lpscr** [comment]
also another think when there is not any check point when finish the train must at less save one last model
 if in save_per_updates  and global_step

https://github.com/SWivid/F5-TTS/blob/99190a03cb473f1d139b7f282e681bbd60c7fd0e/model/trainer.py#L292-L296

when train finish never save the last_model
so i think when i finish the train if not there is any check point i see user train model and use big value for save_per_updates  and global_step and finish the train and never save any checkpoint

self.save_checkpoint(global_step, last=True)

**Comment by SWivid** [comment]
This might cause some problems e.g. 
someone may need previous ckpt with optimizer and scheduler states if he/she find that training went wild and want to go back to some specific point.

From my experience, i'll set a large `save_per_updates` if no need for inter ckpts; or I periodically check for training status and delete unneeded ckpts.

**Comment by lpscr** [comment]
> This might cause some problems e.g. someone may need previous ckpt with optimizer and scheduler states if he/she finds that training went wild and wants to go back to some specific point.
> 
> From my experience, i'll set a large `save_per_updates` if no need for inter ckpts; or I periodically check for training status and delete unneeded ckpts.

ok understand thank you , also one last think if i have only ema can i finetune the model ?

**Comment by SWivid** [comment]
> when train finish never save the last_model

ah yes, that a good point!
will add a saving option before quit
https://github.com/SWivid/F5-TTS/blob/99190a03cb473f1d139b7f282e681bbd60c7fd0e/model/trainer.py#L298

**Comment by SWivid** [comment]
> if i have only ema can i finetune the model

yes for sure, will just initialize the optimizer and scheduler and start from step=0, while loading pretrained model weights (ema basically)

**Comment by lpscr** [comment]
ok i close this now @SWivid 

thank you very much for the reply
